package/MDAnalysis/analysis/lineardensity.py

```diff
@@ -199,10 +210,24 @@ def _conclude(self):
             for key in ['pos', 'pos_std', 'char', 'char_std']:
                 self.results[dim][key] /= self.n_frames
             # Compute standard deviation for the error
-            self.results[dim]['pos_std'] = np.sqrt(self.results[dim][
-                'pos_std'] - np.square(self.results[dim]['pos']))
-            self.results[dim]['char_std'] = np.sqrt(self.results[dim][
-                'char_std'] - np.square(self.results[dim]['char']))
+            # For certain tests in testsuite, floating point imprecision
+            # can lead to negative radicands of tiny magnitude (yielding nan),
+            # rather than actual value of zero. radicand_pos and radicand_char
+            # are therefore tested for closeness to 0 first and changed where
+            # appropriate before the square root is calculated.
+            radicand_pos = self.results[dim][
+                'pos_std'] - np.square(self.results[dim]['pos'])
+            for i in range(len(radicand_pos)):
+                if np.isclose(radicand_pos, np.zeros(self.nbins))[i]:
+                    radicand_pos[i] = 0
+            self.results[dim]['pos_std'] = np.sqrt(radicand_pos)
+
+            radicand_char = self.results[dim][
+                'char_std'] - np.square(self.results[dim]['char'])
+            for i in range(len(radicand_char)):
+                if np.isclose(radicand_char, np.zeros(self.nbins))[i]:
+                    radicand_char[i] = 0
```

Comment:
```suggestion
            radicand_char = np.where(np.isclose(radicand_char, 0), 0, radicand_char)
```
or something on those lines.

Reply:
Or even
```suggestion
            radicand_char[radicand_char < 0] = 0
```